R/sidebar.R

```diff
@@ -218,40 +218,33 @@ layout_sidebar <- function(
 }
 
 sidebar_js_init <- function() {
-  tags$script("data-bslib-sidebar-init" = NA, HTML(
-    "
-    var thisScript = document.querySelector('script[data-bslib-sidebar-init]');
-    thisScript.removeAttribute('data-bslib-sidebar-init');
+  tags$script(
+     "data-bslib-sidebar-init" = NA,
+    "BslibSidebar.initSidebar(document.currentScript)"
```

Comment:
And/or investigate other directions (it would be nice to avoid `document.currentScript` for backwards compatibility reasons)

Reply:
I've reconfigured this so that the script can be included anywhere in the document; as along as it's executed we're okay.

1. The sidebar layout gains a `data-bslib-sidebar-init="{id}"` attribute, with the ID of the sidebar to initialize.
2. We only need to run our initialization code when the sidebar is collapsible, in which case this ID is guaranteed to be present.
3. The initialization script then finds the layout with the data attribute, of which there will be only one, and then initializes it, which removes the initalization data attribute flag.
4. We still use `document.currentScript`, but only to remove that initializaton script from the DOM. Fun fact: you can do that at the start of the script and browsers still will finish running the javascript code in the `<script>` tag.